sdk/wallet-adapter/wallet-kit/src/WalletKitContext.tsx

```diff
@@ -41,26 +39,22 @@ export function WalletKitProvider({
 	disableAutoConnect,
 	features,
 }: WalletKitProviderProps) {
-	const adapters = useMemo(
-		() =>
-			configuredAdapters ?? [
-				new WalletStandardAdapterProvider({ features }),
-				...(enableUnsafeBurner ? [new UnsafeBurnerWalletAdapter()] : []),
-			],
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-		[configuredAdapters],
-	);
-
 	const walletKitRef = useRef<WalletKitCore | null>(null);
 	if (!walletKitRef.current) {
 		walletKitRef.current = createWalletKitCore({
-			adapters,
+			features,
 			preferredWallets,
 			storageAdapter,
 			storageKey,
 		});
 	}
 
+	useEffect(() => {
+		if (!enableUnsafeBurner) return;
+
+		return registerUnsafeBurnerWallet();
```

Comment:
why register the burner wallet in the clean-up part of the effect vs doing the following?

```
if (enableUnsafeBurner) {
  registerUnsafeBurnerWallet();
}
```

Reply:
It’s not registered in cleanup, it’s calling it immediately. the function returns a function to unregister the wallet, which it’ll use as cleanup.

Reply:
Ah I see, yeah it's concise but admittedly a little hard to reason through at first glance (at least for me). I think I've seen similar with the `history.listen` API in react-router. No need to change this since it's kind of a personal preference thing, but sometimes I like the extra explicitness 😄, e.g. 

```
const unregisterBurnerWallet = registerBurnerWallet();
return unregisterBurnerWalet;
```